doc/JSON_INFO.md

```diff
@@ -1883,23 +1883,26 @@ Any Item can be a container. To add the ability to contain things to an item, yo
-    "spoil_multiplier": 1.0,                  // how putting an item in this pocket affects spoilage. less than 1.0 and the item will be preserved longer.
-    "weight_multiplier": 1.0,                 // The items in this pocket magically weigh less inside than outside.  Nothing in vanilla should have a weight_multiplies.
-    "magazine_well": "0 ml",                  // only works if rigid = false, this is the amount of space you can put items in the pocket before it starts expanding
-    "moves": 100,                             // Indicates the number of moves it takes to remove an item from this pocket, assuming best conditions.
-    "fire_protection": false,                 // If the pocket protects the contained items from exploding in a fire or not.  This is for protecting ammo from exploding if the container is tossed into a fire.
-    "watertight": false,                      // can contain liquid
-    "airtight": false,                        // can contain gas
-    "open_container": false,                  // Default is false. If true, the contents of this pocket will spill if this item is placed into another item.
-    "flag_restriction": [ "FLAG1", "FLAG2" ], // items can only be placed into this pocket if they have a flag that matches one of these flags.
-    "rigid": false,                           // Default is false. If false, this pocket's contents contribute to this item's size. If true, they do not.  Think glass jar vs plastic bag: a plastic bag containing nothing takes up almost no space, whereas a glass jar containing nothing takes up as much space as a completely full glass jar. The property magazine_well only works if rigid is false.
-    "holster": false, // if this value is set to true, only one stack of items can be placed inside this pocket, or one item if that item is not count_by_charges.
-    "sealed_data": { "spoil_multiplier": 0.0 } // have anything in sealed_data means the pocket cannot be resealed. Additionally, the sealed version of the pocket will override the unsealed version of the same datatype.
+    "pocket_type": "CONTAINER",       // Typical container pocket. Pockets can also be MAGAZINE.
+    "max_contains_volume": mandatory, // Maximum volume this pocket can hold, totaled among all contained items.  For example "2 L" or "2000 ml" would hold two liters of items.
+    "max_contains_weight": mandatory, // Maximum weight this pocket can hold, totaled among all container items.  For example "6 kg" is about enough to contain a bowling ball.
+    "min_item_volume": "0 ml",        // Minimum volume of item that can be placed into this pocket.  Items smaller than this cannot be placed in the pocket.
+    "max_item_volume": "0 ml",        // Maximum volume of item that can fit through the opening into this pocket.  For example, a 2-liter bottle has a "17 ml" opening.
+    "max_item_length": "0 mm",        // Maximum length of items that can fit in this pocket, by their longest_side.
```

Comment:
i think it's worth mentioning in the documentation that the mandatory member doesn't default to zero, but the default is calculated based on its volume. in this case, it's cbrt( volume ) * sqrt( 2 )

Reply:
Good idea - I found that part in the pocket JSON loading function, but I wonder why `sqrt(2)` is used here. If I've understood right, since we're treating the pocket as a cube, the longest diagonal should be `sqrt(3)`, no?

![image](https://user-images.githubusercontent.com/238395/82733737-35a10900-9cd3-11ea-86f6-3bc9789983fa.png)

Reply:
I've interpreted this to be a kind of "max length that fits the opening", and added tests/documentation to that effect. I can see that using `sqrt(3)` is more of an edge case for infinitely-thin items, and the need to account for an unknown (but usually significant) thickness factor makes `sqrt(2)` more sensible.

The test ensures that a 1-liter pocket with undefined max length (a 1-liter cube 10 cm on each side) allows a `10*sqrt(2) =~ 14 cm` item to fit, but an item with `longest_side: 15 cm` will not.